common/reducers/wallet.ts

```diff
@@ -30,9 +35,21 @@ export interface State {
   hasSavedWalletTokens: boolean;
 }
 
+export type QrTransaction = EthTx;
+export type QrSignature = string;
+export type QrAddress = string;
+
+export interface QrSignatureState {
+  tx: QrTransaction;
+  from: QrAddress;
+  onSignature(signature: QrSignature): void;
+  onCancel(): void;
+}
+
 export const INITIAL_STATE: State = {
   inst: null,
   config: null,
+  signViaQr: null,
```

Comment:
IMO (Would definitely like additional opinions) this should probably be its own reducer with its own set of actions / sagas. I think this is pretty separate from the other wallet processes, and is specific to only the parity signer.

Reply:
I definitely see maintainability improvements with such a separation of concerns 👍